icu4c/source/i18n/unicode/formattednumber.h

```diff
@@ -190,6 +191,9 @@ class U_I18N_API FormattedNumber : public UMemory, public FormattedValue {
 
     // To give C API access to internals
     friend struct impl::UFormattedNumberImpl;
+
+    // To give access to the data pointer for non-heap allocation
+    friend class icu::SimpleDateFormat;
```

Comment:
I'd kind of like @sffc to weigh in on whether he's okay with adding these `friend` declarations to `SimpleNumberFormatter` and `FormattedNumber`.

Reply:
Fine with me